Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,826 @@
+        let opts = ["find-module", m, "--simple-output"] ++ ["--global", "--user"] ++ optsForGhcPkg []
+        gmLog GmDebug "_ghcPkgFindModule" $ strDoc $ "ghc-pkg " ++ show opts
+
+        x <- runCmd rp "ghc-pkg" opts
+
+        -- gmLog GmDebug "" $ strDoc $ "_ghcPkgFindModule stdout: " ++ show output
+        -- gmLog GmDebug "" $ strDoc $ "_ghcPkgFindModule stderr: " ++ show err
+        return $ case x of
+                    Just x' -> join $ (Safe.lastMay . words) <$> (Safe.lastMay . lines) x'
+                    Nothing -> Nothing
+
+    -- | Call @cabal sandbox hc-pkg@ to find the package the provides a module.
+    -- hcPkgFindModule :: String -> IO (Maybe String)
+    hcPkgFindModule rp m = do
+        let opts = ["sandbox", "hc-pkg", "find-module", m, "--", "--simple-output"]
+
+        x <- runCmd rp "cabal" opts
+
+        -- gmLog GmDebug "" $ strDoc $ "hcPkgFindModule stdout: " ++ show output
+        -- gmLog GmDebug "" $ strDoc $ "hcPkgFindModule stderr: " ++ show err
+        return $ case x of
+                    Just x' -> join $ (Safe.lastMay . words) <$> (Safe.lastMay . lines) x'
+                    Nothing -> Nothing
+
+    -- | Call @stack exec ghc-pkg@ to find the package the provides a module.
+    -- stackGhcPkgFindModule :: String -> IO (Maybe String)
+    stackGhcPkgFindModule rp m = do
+        let opts = ["exec", "ghc-pkg", "find-module", m, "--", "--simple-output"]
+
+        x <- runCmd rp "stack" opts
+
+        -- gmLog GmDebug "" $ strDoc $ "stackGhcPkgFindModule stdout: " ++ show output
+        -- gmLog GmDebug "" $ strDoc $ "stackGhcPkgFindModule stderr: " ++ show err
+        return $ case x of
+                    Just x' -> join $ (Safe.lastMay . words) <$> (Safe.lastMay . lines) x'
+                    Nothing -> Nothing
+
+ghcPkgHaddockUrl
+    :: forall m. (GmLog m, GmOut m, MonadIO m)
+    => FilePath
+    -> (FilePath -> [String] -> String -> IO String)
+    -> [GhcPkgDb]
+    -> String
+    -> m (Maybe String)
+ghcPkgHaddockUrl ghcPkg readProc pkgDbStack p = do
+    gmLog GmDebug "ghcPkgHaddockUrl" $ strDoc p
+
+    let p' = case splitOn "@" p of
+                [p0, _] -> p0
+                _       -> p
+
+    hout <- liftIO $ readProc ghcPkg (toDocDirOpts p' pkgDbStack) ""
+    return $ Safe.lastMay $ words $ reverse . dropWhile (== '\n') . reverse $ hout
+  where
+    -- This fails unless we have --global and --user, unlike
+    -- pkgDoc elsewhere in ghc-mod.
+    toDocDirOpts pkg dbs = ["field", pkg, "haddock-html", "--global", "--user"] ++ ghcPkgDbStackOpts dbs
+
+ghcPkgHaddockInterface
+    :: forall (m :: * -> *).  MonadIO m
+    => FilePath
+    -> (FilePath -> [String] -> String -> IO String)
+    -> [GhcPkgDb]
+    -> String
+    -> m (Maybe String)
+ghcPkgHaddockInterface ghcPkg readProc pkgDbStack p = do
+    hout <- liftIO $ readProc ghcPkg (toHaskellInterfaces p pkgDbStack) ""
+    return $ Safe.lastMay $ words $ reverse . dropWhile (== '\n') . reverse $ hout
```

Comment:
Either `.` or `$` not both please, also redundant `$` at the end of the line. Maybe you should run `ghc-mod hlint` over the source ;)